internal/ethereum/blocklistener.go

```diff
@@ -140,6 +144,28 @@ func (bl *blockListener) listenLoop() {
 		update := &ffcapi.BlockHashEvent{GapPotential: gapPotential}
 		var notifyPos *list.Element
 		for _, h := range blockHashes {
+			if len(h) != 32 {
```

Comment:
Sorry @SamMayWork I couldn't understand in the end the problem and the solution. Could do with some help.

32 is the length of a "standard" Ethereum block hash. So if we get a standard one, it will be 32 bytes in length.

You are working with a EVM implementation against a ledger that uses a 48 byte block hash - so in that case, the length here would be 48.

Why would we truncate that? Surely we just need to acommodate that it's 48 bytes, and there will be some number of places in the code that need to be relaxed to avoid the assumption of 32 bytes. Maybe for example there are DB migration tables etc.

Reply:
> Why would we truncate that? Surely we just need to acommodate that it's 48 bytes, and there will be some number of places in the code that need to be relaxed to avoid the assumption of 32 bytes. Maybe for example there are DB migration tables etc.

There was a brief [Discord discussion](https://discord.com/channels/905194001349627914/928377875827154984/1219628950393847860) on this, although in retrospect I don't think we considered what other alternative approaches we could use here.

The reason for using truncation here instead of changing the code everywhere to permit the change in header size was purely to reduce the scope of the changes required. I think the view was that Hedera _is_ EVM compatible but also a little bit weird, and it might make more sense to truncate the hashes so there's no need to change existing code for a single use-case.

We could use this PR to permit non-standard block header sizes everywhere if that's something strategically we think EVM Connect should be able to do.

Reply:
```suggestion
			if len(h) == 48 {
```